dna/dna.go

```diff
@@ -1,38 +1,184 @@
+// Package dna provides functionality to create and manipulate DNA sequences.
 package dna
 
-import ()
+import "strings"
 
+// Base stores a single nucleotide as a byte.
 type Base byte
 
-// TODO: change these names so that all variables can be seen
-// from outside.  Maybe BaseA, Basea, etc
 const (
-	A   Base = 0
-	C   Base = 1
-	G   Base = 2
-	T   Base = 3
-	N   Base = 4
-	a   Base = 5
-	c   Base = 6
-	g   Base = 7
-	t   Base = 8
-	n   Base = 9
-	Gap Base = 10
-	Dot Base = 11
+	A      Base = 'A'
+	C      Base = 'C'
+	G      Base = 'G'
+	T      Base = 'T'
+	N      Base = 'N'
+	LowerA Base = 'a'
+	LowerC Base = 'c'
+	LowerG Base = 'g'
+	LowerT Base = 't'
+	LowerN Base = 'n'
+	Gap    Base = '-'
+	Dot    Base = '.'
+	Nil    Base = '*'
 )
 
-func CreateAllGaps(numGaps int64) []Base {
+// CreatAllGaps creates a DNA sequence of Gap with length of numGaps
+func CreateAllGaps(numGaps int) []Base {
 	answer := make([]Base, numGaps)
-	for i := 0; i < len(answer); i++ {
+	for i := range answer {
 		answer[i] = Gap
 	}
 	return answer
 }
 
-func CreateAllNs(numGaps int64) []Base {
-	answer := make([]Base, numGaps)
-	for i := 0; i < len(answer); i++ {
+// CreatAllN creates a DNA sequence of N with length of numGaps
+func CreateAllNs(numN int) []Base {
+	answer := make([]Base, numN)
+	for i := range answer {
 		answer[i] = N
 	}
 	return answer
 }
+
+//BaseToString converts a DNA base to a string by casting a BaseToRune result to a string.
+func BaseToString(b Base) string {
+	return string(b)
+}
+
+//StringToBases parses a string into a slice of DNA bases
+func StringToBases(s string) []Base {
+	return []Base(s)
```

Comment:
I am worried about a lack of input checking here where almost any string would get turned into bases, but the []Base would have strange values in it.  Before I think we had error checking here to make sure someone didn't give us a protein sequence or whatever else they might do.